doc/3d.rst

```diff
@@ -54,12 +58,233 @@ The 3d Dependent Data Description language
 
    3d-lang
 
-The 3d frontend
----------------
+Download and install
+--------------------
+
+We provide binary (amd64 only) packages for Windows and Linux.
+
+1. Download the binary package corresponding to your platform, from
+   the `latest EverParse/3d release
+   <https://github.com/project-everest/everparse/releases>`_.
+
+2. Extract it (Windows: ``unzip``; Linux: ``tar xzf``) anywhere you
+   like. No need to configure it anyhow.
+
+Then, everything is contained in a ``everparse`` directory, whose
+actual entrypoint executable is ``everparse.bat`` (for Windows) or
+``everparse.sh`` (for Linux.)
+
+Binary packages already contain all dependencies required by
+EverParse, such as F*, KReMLin and z3, so no need to install anything
+beforehand.
+
+The Windows binary package contains a `binary snapshot of clang-format
+<https://llvm.org/builds/>`_ to pretty-print the produced ``.c`` and
+``.h`` files. On Linux, you may want to install clang-format (more
+info at the `LLVM download page
+<https://releases.llvm.org/download.html>`_), but this is optional.
+
+The Windows binary package is fully native, in the sense that it does
+not require any POSIX emulation layer such as Cygwin (contrary to the
+sources) or WSL.
+
+Run
+---
+
+On Windows: ``everparse.bat`` [options] input_files
+
+On Linux: ``everparse.sh`` [options] input_files
+
+Options are treated from left to right:
+
+* ``--odir output_directory_name``
+
+  Tell EverParse to write all files to the directory
+  ``output_directory_name``. This directory must already exist
+  (EverParse will not create it.) The default is ``.`` (current
+  directory)
+
+* ``--batch``
+
+  Verify the produced F* ``.fst`` and ``.fsti`` files and generate C
+  ``.c`` and ``.h`` files. This is toggled by default.
+
+* ``--no_batch``
+
+  Do not verify the produced F* ``.fst`` and ``.fsti`` files or
+  generate C ``.c`` and ``.h`` files.
+
+* ``--clang_format``
+
+  Call ``clang-format`` on extracted .c/.h files. This option
```

Comment:
this is to shorten the command line when invoking `3d`. Indeed `--clang_format` only makes sense with `--batch` on because `.c`/`.h` files are produced only with `--batch` (otherwise only `.fst*` files are produced)